receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/accumulator.go

```diff
@@ -34,34 +34,61 @@ func (acc *metricDataAccumulator) getMetricsData(containerStatsMap map[string]*C
 	taskResource := taskResource(metadata)
 
 	for _, containerMetadata := range metadata.Containers {
+
+		containerResource := containerResource(containerMetadata)
+		taskResource.Attributes().ForEach(func(k string, av pdata.AttributeValue) {
+			containerResource.Attributes().Upsert(k, av)
+		})
+
 		stats, ok := containerStatsMap[containerMetadata.DockerID]
-		if !ok || stats == nil {
-			continue
-		}
 
-		containerMetrics := getContainerMetrics(stats, logger)
-		if containerMetadata.Limits.Memory != nil {
-			containerMetrics.MemoryReserved = *containerMetadata.Limits.Memory
+		if ok && !isEmptyStats(stats) {
+			containerMetrics := convertContainerMetrics(stats, logger, containerMetadata)
+			acc.accumulate(convertToOTLPMetrics(ContainerPrefix, containerMetrics, containerResource, timestamp))
+			aggregateTaskMetrics(&taskMetrics, containerMetrics)
+			calculatTaskLevelLimit(&taskMetrics, metadata)
+			acc.accumulate(convertToOTLPMetrics(TaskPrefix, taskMetrics, taskResource, timestamp))
+		} else {
+			acc.accumulate(convertContainerResource(containerResource))
 		}
+	}
+}
 
-		if containerMetadata.Limits.CPU != nil {
-			containerMetrics.CPUReserved = *containerMetadata.Limits.CPU
-		}
+func (acc *metricDataAccumulator) accumulate(rms pdata.ResourceMetricsSlice) {
+	md := pdata.Metrics(rms)
 
-		if containerMetrics.CPUReserved > 0 {
-			containerMetrics.CPUUtilized = (containerMetrics.CPUUtilized / containerMetrics.CPUReserved)
-		}
+	acc.mds = append(acc.mds, md)
+}
 
-		containerResource := containerResource(containerMetadata)
-		taskResource.Attributes().ForEach(func(k string, av pdata.AttributeValue) {
-			containerResource.Attributes().Upsert(k, av)
-		})
+func isEmptyStats(stats *ContainerStats) bool {
+	return stats == nil || stats.ID == ""
+}
+
+func convertContainerResource(containerResource pdata.Resource) pdata.ResourceMetricsSlice {
+	rms := pdata.NewResourceMetricsSlice()
+	rms.Resize(1)
+	rm := rms.At(0)
+	containerResource.CopyTo(rm.Resource())
+	return rms
+}
+
+func convertContainerMetrics(stats *ContainerStats, logger *zap.Logger, containerMetadata ContainerMetadata) ECSMetrics {
+	containerMetrics := getContainerMetrics(stats, logger)
+	if containerMetadata.Limits.Memory != nil {
+		containerMetrics.MemoryReserved = *containerMetadata.Limits.Memory
+	}
 
-		acc.accumulate(convertToOTLPMetrics(ContainerPrefix, containerMetrics, containerResource, timestamp))
+	if containerMetadata.Limits.CPU != nil {
+		containerMetrics.CPUReserved = *containerMetadata.Limits.CPU
+	}
 
-		aggregateTaskMetrics(&taskMetrics, containerMetrics)
+	if containerMetrics.CPUReserved > 0 {
+		containerMetrics.CPUUtilized = (containerMetrics.CPUUtilized / containerMetrics.CPUReserved)
 	}
+	return containerMetrics
+}
 
+func calculatTaskLevelLimit(taskMetrics *ECSMetrics, metadata TaskMetadata) {
```

Comment:
Is is possible to rename it? We are not calculating taskLevelLimit here. We are overriding with Task level value if they are available.

Reply:
OK, maybe overrideTaskLevelLmit?

Reply:
OverrideWithTaskLevelLimit?